### PR TITLE
fix curl write callback to not exit

### PR DESCRIPTION
**Pull request by wolfv**
cc @Klaim

### CODE REVIEW

File: libmamba/src/core/fetch.cpp [hunk 1]
```diff
@@ -344,6 +344,30 @@ namespace mamba
...
+        {
+            case CURLE_ABORTED_BY_CALLBACK:
+            case CURLE_BAD_FUNCTION_ARGUMENT:
+            case CURLE_CONV_REQD:
+            case CURLE_COULDNT_RESOLVE_PROXY:
+            case CURLE_FILESIZE_EXCEEDED:
+            case CURLE_INTERFACE_FAILED:
+            case CURLE_NOT_BUILT_IN:
+            case CURLE_OUT_OF_MEMORY:
+            // See RhBug: 1219817
```
Comment (Klaim) [inline libmamba/src/core/fetch.cpp:357]: Is this a reference from librepo?

File: libmamba/src/core/fetch.cpp [hunk 2]
```diff
@@ -391,7 +415,7 @@ namespace mamba
             {
                 LOG_ERROR << "Could not open file for download " << s->m_filename << ": "
                           << strerror(errno);
-                exit(1);
+                return size * nmemb + 1;
```
Comment (Klaim) [inline libmamba/src/core/fetch.cpp:418]: This is equivalent to `(size * nmemb) + 1`, is this what you intended? (it's not clear for me what the `1` is).

File: libmamba/src/core/fetch.cpp [hunk 3]
```diff
@@ -400,7 +424,7 @@ namespace mamba
         if (!s->m_file)
         {
             LOG_ERROR << "Could not write to file " << s->m_filename << ": " << strerror(errno);
-            exit(1);
+            return size * nmemb + 1;
```
Comment (Klaim) [inline libmamba/src/core/fetch.cpp:427]: Same question

**Review by Klaim — APPROVED** [review]
I mostly have minor questions, LGTM otherwise.

File: libmamba/src/core/fetch.cpp (hunk 1 above)
Comment (wolfv) [reply libmamba/src/core/fetch.cpp:357]: yep

File: libmamba/src/core/fetch.cpp (hunk 2 above)
Comment (wolfv) [reply libmamba/src/core/fetch.cpp:418]: we just need to return a "size different from size * nmemb". We could always return `0` but it's possible (unlikely though) that we download an empty file with `0` bytes (the curl manual mentions this). I chose to just randomly add `1`. Maybe I should add a comment ...

File: libmamba/src/core/fetch.cpp (hunk 3 above)
Comment (wolfv) [reply libmamba/src/core/fetch.cpp:427]: same answer :)

**Pull request merged**